packages/tools/src/store/SynchronizerManager/Synchronizer.ts

```diff
@@ -211,13 +212,20 @@ class Synchronizer {
         if (targetIsSource) {
           continue;
         }
-
-        this._eventHandler(this, sourceViewport, targetViewport, sourceEvent);
+        promises.push(
```

Comment:
I don't know how to avoid this synchronization - the basic synchronizer design relies on being able to turn off secondary notifications so that only a single change applies, and if this doesn't use async, then you get an event from the target viewport, which changs the source viewport, which sends an event etc.